### PR TITLE
Server student cap

### PR DESCRIPTION
**Pull request by msakr21**
This PR adds:

1- A final confirmation/application received page with information about the session and a link* to the library.

2- Server side checks to prevent a prospective student from being able to join a session that's full.

Note(s):

*Link is currently hard coded in a dictionary with the location being the key. Will change in future depending on further discussions. Also, link is a shortened (probably bitly) link that seems to fail tests on occasion, so may need to put the long link in the dictionary instead as the shortened one may involve an API that possibly gets overloaded.

### CODE REVIEW

File: config/credentials.yml.enc
Comment (bkeener7) [inline config/credentials.yml.enc:1]: Still curious on where this is from. I don't seem to get them on my PRs. Might have something to do with new pages?

**Review by bkeener7 — APPROVED** [review]
Huge PR! Tests look fleshed out. I see some opportunities to DRY up some code, but they are minor. I'll need to get more updates from you on the failing test. I've found tests that occasionally fail should always be avoided. We'll need to see if there are alternative solutions.

File: config/credentials.yml.enc
Comment (msakr21) [reply config/credentials.yml.enc:1]: There's a terminal command that needed to be run to generate credentials for render. That file is not it though, but I do remember when helping others having something like this file also come up which we'd just delete as it's not needed.

It should be removed at this point though (that was the update with this PR).

**Comment by msakr21** [comment]
Yeah. I'm pretty happy with the edge case I thought of wherein someone could access the form only for another to submit for the last spot before they're done which would have led to going over the limit! 

Beside that, it's just the one test that seems to fail haphazardly and it doesn't seem to be because Capybara needs a bit of time before doing checks (which was the case with a couple of other tests hence my adding a wait: 2 to them). I suspect the short links are set up through an bitly api endpoint and so repeated calls probably give it trouble? I could throw in a wait time and see if it does anything, but because it is consistent in its failure I'm not sure if I'll be able to tell with certainty if it's failing or not. Could stub it, maybe?

**Pull request merged**